webviz_subsurface_components/VectorCalculatorWrapper.py

```diff
@@ -0,0 +1,205 @@
+                )
+
+            invalid_var_chars = [
+                elm for elm in variables if not re.search("[a-zA-Z]{1}", elm)
+            ]
+            if len(invalid_var_chars) > 0:
+                message = (
+                    "Invalid variable characters:"
+                    if len(invalid_var_chars) > 1
+                    else "Invalid variable character:"
+                )
+                raise Exception(message + f" {invalid_var_chars}")
+
+            # Evaluate to ensure valid expression (not captured by parse() method)
+            # - Parser allow assignment of function to variable, e.g. parse("f(x)").evaluate({"f":np.sqrt, "x":2})
+            # - Assign value to variables and evaluate to ensure valid expression
+            evaluation_values = np.ones(len(variables))
+            evaluation_dict = dict(zip(variables, evaluation_values))
+            parsed_expr.evaluate(evaluation_dict)
+
+            return expression
+
+    @staticmethod
+    def external_parse_data(expression: ExpressionInfo) -> ExternalParseData:
+        try:
+            expression_string = VectorCalculatorWrapper.parse_expression(
+                expression["expression"]
+            )
+            variables = VectorCalculatorWrapper.parser.parse(
+                expression_string
+            ).variables()
+
+            return {
+                "expression": expression_string,
+                "id": expression["id"],
+                "variables": variables,
+                "isValid": True,
+                "message": "",
+            }
+        except Exception as e:
+            return {
+                "expression": expression["expression"],
+                "id": expression["id"],
+                "variables": [],
+                "isValid": False,
+                "message": "" if len(expression["expression"]) <= 0 else str(e),
+            }
+
+    def validate_expression(expression: ExpressionInfo) -> bool:
+        try:
+            VectorCalculatorWrapper.parse_expression(expression["expression"])
+        except:
+            return False
+        return True
+
+    @staticmethod
+    def evaluate_expression(
+        expression: str, values: Dict[str, pd.Series]
+    ) -> Union[pd.Series, None]:
+        result: pd.Series = pd.Series()
+
+        # Ensure variables in expression
+        invalid_variables = [var for var in values if var not in expression]
+        if len(invalid_variables) > 0:
+            warnings.warn(
+                f"Variables {invalid_variables} is not present in expression '{expression}'"
+            )
+            return result
+        try:
+            expression = VectorCalculatorWrapper.parse_expression(expression)
+            parsed_expr = VectorCalculatorWrapper.parser.parse(expression)
+            evaluated_expr: list = parsed_expr.evaluate(values)
+            result = pd.Series(evaluated_expr)
+        except ValueError as e:
+            result = None
+        return result
```

Comment:
From video discussion:
```suggestion
    def evaluate_expression(
        expression: str, values: Dict[str, np.ndarray]
    ) -> Union[np.ndarray, None]:
        # result: pd.Series = pd.Series()
        # Ensure variables in expression
        invalid_variables = [var for var in values if var not in expression]
        if len(invalid_variables) > 0:
            warnings.warn(
                f"Variables {invalid_variables} is not present in expression '{expression}'"
            )
            return None
        try:
            expression = VectorCalculatorWrapper.parse_expression(expression)
            parsed_expr = VectorCalculatorWrapper.parser.parse(expression)
            return parsed_expr.evaluate(values)
        except ValueError as e:
            return None
```